### PR TITLE
Update readme for register processor

### PR DESCRIPTION
**Pull request by Alex7Li**
Update the README file for the register processor command. This commit requires the register processor command to have been made #29 and is built off of it. The diff starts at commit 8824743 - update the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -386,27 +392,83 @@ function listener(model, message) {
...
+
+// Important: to access the model using this,
+// registered processors must not be arrow functions.
+helper.registerProcessor('copy', function() {
+  const sum = this.get('sum');
+  // We could also do this.set('ans', sum), but changing
+  // the model inside of a registered processor is discouraged.
+  return {ans: sum};
+});
+
```
Comment (zikas) [inline README.md:449]: nit: extra line

File: README.md [hunk 2]
```diff
@@ -298,11 +301,14 @@ values in the abstract data model.
 So far, we've seen that objects (messages) can be pushed onto the dataLayer, as well as arrays 
 (command arrays). Pushing a function onto the dataLayer will also allow you to update the abstract 
 data model, but with custom code. This technique has the added benefit of being able to handle 
-return values of any native method calls made from within the function.
-
-When a function is processed, it will be executed in the context of the abstract data model. The 
-value of "this" will be an interface that represents the current abstract data model. This 
-interfact will provide two APIs: get(key) and set(key, value). The following examples demonstrate 
+return values of any native method calls made from within the function. When a function is processed, the 
+value of "this" will be the abstract data model interface, described below.
```
[the remark this answers is not in the document]
Comment (Alex7Li) [reply README.md:305]: I'm not sure what you mean by this? I added a line break between the sections

[the remark this answers is not in the document]
Comment (Alex7Li) [reply README.md:305]: Yep

File: README.md [hunk 3]
```diff
@@ -298,11 +301,14 @@ values in the abstract data model.
...
+#### The Abstract Data Model Interface
+To safely access the abstract data model from within a custom method, an
+API with a getter and setter is provided. get(key) will get a key of the model,
+and set(key, value) will create or overwrite the given key with the new value.
+
+The following examples demonstrate
```
[the remark this answers is not in the document]
Comment (Alex7Li) [reply README.md:311]: I went and looked more carefully at the whole file to codify everythin

File: README.md (hunk 1 above)
Comment (Alex7Li) [reply README.md:449]: Removed

**Pull request merged**